pgxn/neon/control_plane_connector.c

```diff
@@ -857,16 +956,37 @@ NeonProcessUtility(
 								dest,
 								qc);
 	}
+	if (sudo)
+		switch_to_original_role();
 }
 
 void
 InitControlPlaneConnector()
 {
 	PreviousProcessUtilityHook = ProcessUtility_hook;
 	ProcessUtility_hook = NeonProcessUtility;
+
+    next_needs_fmgr_hook = needs_fmgr_hook;
+	needs_fmgr_hook = neon_needs_fmgr_hook;
+
+	next_fmgr_hook = fmgr_hook;
+	fmgr_hook = neon_fmgr_hook;
+
 	RegisterXactCallback(NeonXactCallback, NULL);
 	RegisterSubXactCallback(NeonSubXactCallback, NULL);
 
+	DefineCustomBoolVariable(
+							 "neon.disable_event_triggers_for_superuser",
```

Comment:
I have a hard doing mental negation for variables that has inverted meaning? Wouldn't it be easier with `allow_event_triggers_for_superuser`?

Reply:
Done